Prox/Prox/Utilities/OpenInHelper.swift

```diff
@@ -0,0 +1,168 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+import Foundation
+import MapKit
+
+private enum Schemes: String {
+    case yelp = "www.yelp.com"
+    case tripAdvisor = "www.tripadvisor.co.uk"
+    case wikipedia = "en.wikipedia.org"
+}
+
+struct OpenInHelper {
+
+    static let gmapsAppSchemeString: String = "comgooglemaps://"
+    static let gmapsWebSchemeString: String = "https://www.google.com/maps"
+    static let appleMapsSchemeString: String = "http://maps.apple.com/"
+    static let yelpAppURLSchemeString: String = "yelp://"
+    static let tripAdvisorAppURLSchemeString: String = "tripadvisor://"
+    static let wikipediaAppURLSchemeString: String = "wikipedia://"
+
+    static func open(url: URL) -> Bool {
```

Comment:
yeah, I understand. Personally, I found the way I did it very clear and fits the way I think about things, but I see where you are coming from. Also as the class grew and I kept refactoring as I added more functionality I realise that I let things get a little tangled. Have cleared it up now.